src/dos/dos_locale.cpp

```diff
@@ -0,0 +1,2904 @@
+		locale_period = config.locale_period;
+	}
+
+	assert(country_info.locale_info.count(locale_period));
+	const auto& source      = country_info.locale_info.at(locale_period);
+	const auto& destination = dos.tables.country;
+
+	// Set reserved/undocumented values to 0's
+
+	memset(&destination[InfoOffsetReserved], 0, ReservedAreaSize);
+
+	// Set time/date/number/list format
+
+	size_t offset = InfoOffsetTimeFormat;
+	destination[offset] = static_cast<uint8_t>(source.time_format);
+
+	offset = InfoOffsetTimeSeparator;
+	destination[offset + 0] = static_cast<uint8_t>(source.time_separator);
+	destination[offset + 1] = 0;
+
+	offset = InfoOffsetDateFormat;
+	destination[offset + 0] = static_cast<uint8_t>(source.date_format);
+	destination[offset + 1] = 0;
+
+	offset = InfoOffsetDateSeparator;
+	destination[offset + 0] = static_cast<uint8_t>(source.date_separator);
+	destination[offset + 1] = 0;
+
+	offset = InfoOffsetThousandsSeparator;
+	destination[offset + 0] = static_cast<uint8_t>(source.thousands_separator);
+	destination[offset + 1] = 0;
+
+	offset = InfoOffsetDecimalSeparator;
+	destination[offset + 0] = static_cast<uint8_t>(source.decimal_separator);
+	destination[offset + 1] = 0;
+
+	offset = InfoOffsetListSeparator;
+	destination[offset + 0] = static_cast<uint8_t>(source.list_separator);
+	destination[offset + 1] = 0;
+
+	// Set currency symbol and format
+
+	assert(source.currency_code.size() < MaxCurrencySymbolLength);
+	memset(&destination[InfoOffsetCurrencySymbol], 0, MaxCurrencySymbolLength + 1);
+
+	bool found = false;
+	for (const auto& candidate_utf8 : source.currency_symbols_utf8) {
+		std::string candidate = {};
+
+		// Check if the currency can be converted to current code page
+
+		if (!utf8_to_dos(candidate_utf8, candidate, UnicodeFallback::Null)) {
+			continue;
+		}
+		if (candidate.length() > MaxCurrencySymbolLength) {
+			continue;
+		}
+
+		found = true;
+		for (const auto character : candidate) {
+			if (character == 0) {
+				found = false;
+				break;
+			}
+		}
+
+		if (found) {
+			memcpy(&destination[InfoOffsetCurrencySymbol],
+			       candidate.c_str(),
+			       candidate.length());
+			break;
+		}
+	}
+
+	offset = InfoOffsetCurrencyFormat;
+	destination[offset] = static_cast<uint8_t>(source.currenct_format);
+
+	if (!found) {
+		// Fallback - use the currency code instead
+		memcpy(&destination[InfoOffsetCurrencySymbol],
+		       source.currency_code.c_str(),
+		       source.currency_code.length());
+
+		// Force separation between symbol and amount
+		bit::set(destination[offset], bit::literals::b0);
+	}
+
+	offset = InfoOffsetCurrencyPrecision;
+	destination[offset] = source.currency_precision;
+
+	// Mark locale as generated
+
+	config.is_locale_generated = true;
+
+	// If locale changed, log it
+
+	static bool already_logged = false;
+
+	static Country logged_country            = {};
+	static LocalePeriod logged_real_period   = {};
+	static LocalePeriod logged_config_period = {};
```

Comment:
Can't we turn this into a class to get rid of the statics?

Reply:
IMHO - not worth it. Especially now, when this code went to a separate function.